policy-controller/src/admission.rs

```diff
@@ -418,51 +426,76 @@ impl Validate<ServerAuthorizationSpec> for Admission {
 #[async_trait::async_trait]
 impl Validate<HttpRouteSpec> for Admission {
     async fn validate(self, _ns: &str, _name: &str, spec: HttpRouteSpec) -> Result<()> {
-        let targets_server = spec.inner.parent_refs.iter().flatten().any(|parent_ref| {
-            if let Some(p) = parent_ref.group.as_deref() {
-                if let Some(k) = parent_ref.kind.as_deref() {
-                    return p.eq_ignore_ascii_case("policy.linkerd.io")
-                        && k.eq_ignore_ascii_case("server");
-                }
-            }
-            false
-        });
+        // The validation for the policy.linkerd.io HTTPRoute type is much
+        // simpler than for the Gateway API version: the route must target a
+        // `Server` resource.
+        //
+        // We don't have to do any validation that unsupported filters aren't
+        // present, because Linkerd's HTTPRoute CRD doesn't include those
+        // filters at all.
+        if !parent_refs_target_server(&spec.inner) {
```

Comment:
I think we want to be more strict here.  Currently this accepts if ANY of the parent refs are a Server.  I think we want to accept if ALL parents are Servers.

Reply:
done in [bac79cc](https://github.com/linkerd/linkerd2/pull/8949/commits/bac79cc75da8c03389a1efcb7fb2a0ca413f5159)!

Reply:
Eh, we'd want to admit gateway resources that target other things; but since we're about to remove that, it's fine to be more strict.